android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,167 @@
+                    receivedPayment = pd.payment
+
+                    // Push back shutdown by SHUTDOWN_DELAY_MS for payments synced event
+                    pushbackShutdown()
+                }
+
+                is BreezEvent.Synced -> {
+                    Logger.tag(TAG).info { "Got synced event" }
+                    receivedPayment?.let { shutdown() }
+                }
+
+                else -> {}
+            }
+        }
+
+        private fun handleReceivedPayment(
+            bolt11: String,
+            paymentHash: String,
+            amountMsat: ULong?,
+        ) {
+            Logger.tag(TAG)
+                .info { "Received payment. Bolt11:${bolt11}\nPayment Hash:${paymentHash}" }
+            val amountSat = (amountMsat ?: ULong.MIN_VALUE) / 1000u
+            notifyPaymentReceived(applicationContext, amountSat = amountSat)
+        }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        Logger.tag(TAG).debug { "Creating Breez foreground service..." }
+        registerNotificationChannels(applicationContext)
+        Logger.tag(TAG).debug { "Breez foreground service created." }
+    }
+
+    // =========================================================== //
+    //                      SERVICE LIFECYCLE                      //
+    // =========================================================== //
+
+    override fun onBind(intent: Intent): IBinder? {
+        return null
+    }
+
+    private val shutdownHandler = Handler(Looper.getMainLooper())
+    private val shutdownRunnable: Runnable = Runnable {
+        Logger.tag(TAG).debug { "Reached scheduled shutdown..." }
+        shutdown()
+    }
+
+    /** Stop the service */
+    private fun shutdown() {
+        Logger.tag(TAG).info { "Shutting down Breez foreground service" }
+        stopForeground(STOP_FOREGROUND_REMOVE)
+        stopSelf()
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        val intentDetails = "[ intent=$intent, flag=$flags, startId=$startId ]"
+        Logger.tag(TAG).debug { "Start Breez foreground service from intent $intentDetails" }
+
+        // Connect to SDK if source intent has data message with valid payload
+        connectSdkIfNeeded(intent)
+
+        // Push back shutdown by SHUTDOWN_DELAY_MS after connecting to the SDK
+        pushbackShutdown()
+        return START_NOT_STICKY
+    }
+
+    /** Checks whether if the data payload is of type payment_received & has a payment hash */
+    private fun connectSdkIfNeeded(intent: Intent?) {
+        intent?.remoteMessage?.run {
```

Comment:
It seems cleaner to check this on the higher level (onStartCommand) and then shutdown in case the message is not recognized.
Right now the process keeps running until the delay ends.

Reply:
Done 79d7566aebf9ea00d6c65143b6a019fbb72f7a5c